anyway/parsers/infographics_data_cache_updater.py

```diff
@@ -13,6 +13,50 @@
 db = SQLAlchemy(app)
 
 
+def is_cache_eligible(news_flash):
+    return (
+        news_flash.accident
+        and news_flash.resolution in (["כביש בינעירוני"])
+        and news_flash.road_segment_name is not None
+    )
+
+
+def is_in_cache(nf):
+    return (
+        len(CONST.INFOGRAPHICS_CACHE_YEARS_AGO)
+        == db.session.query(InfographicsDataCache)
+        .filter(InfographicsDataCache.news_flash_id == nf.get_id())
+        .count()
+    )
+
+
+def add_news_flash_to_cache(news_flash):
+    if not is_cache_eligible(news_flash):
+        logging.debug(
+            f"add_news_flash_to_cache: news flash does not qualify:{news_flash.serialize()}"
+        )
+        return
+    try:
+        db.get_engine().execute(
+            InfographicsDataCache.__table__.insert(),  # pylint: disable=no-member
+            [
+                {
+                    "news_flash_id": news_flash.get_id(),
+                    "years_ago": y,
+                    "data": anyway.infographics_utils.create_infographics_data(
+                        news_flash.get_id(), y
+                    ),
+                }
+                for y in CONST.INFOGRAPHICS_CACHE_YEARS_AGO
+            ],
+        )
+    except Exception as e:
```

Comment:
Why catch `Exception`? What are the exceptions meant to be caught, and why not raise them?

Reply:
I thought we do not want that the add-to-cache activity will interrupt the process of adding new news-flash items. If we do, then the exception should not be catched.

Reply:
Hmmm. If it is a side activity, maybe it should (ideally, not as a request for this PR) be performed in an independent process? Then we wouldn't care if it crashed.

I mean, we could have a caching process, and communicate with it using a queue, which will give better failure isolation. Or maybe that would be over-engineering?

Reply:
I agree with @ziv17 - we don't want to interrupt the process of adding new news-flash items at this point.
I also agree with @elazarg - it will be performed as an independent process in the future, I will open relevant issue for this separation in the code (I think we'll do it with airflow).